src/transformer/cli.js

```diff
@@ -53,28 +53,69 @@ export default async ({name, yargsOptions, callback}) => {
 		process.exit(-1);
 	}
 
-	const spinner = ora('Transforming records').start();
+	const spinner = ora(`Transforming${args.validate ? ' and validating' : ''}${args.fix ? ' and fixing' : ''} records`).start();
+
 	const options = {
 		stream: fs.createReadStream(args.file),
-		args,
-		spinner,
-		handleRecordsOutput
+		args
 	};
-	await callback(options);
 
-	function handleRecordsOutput(records) {
+	const TransformClient = callback(options);
+	let succesRecordArray = [];
+	let failedRecordsArray = [];
+	let pendingPromises = [];
+
+	await new Promise(resolve => {
+		TransformClient
+			.on('end', async () => {
+				Promise.all(pendingPromises);
+				resolve();
+			})
+			.on('error', errorEvent)
+			.on('record', async payload => {
+				pendingPromises.push(recordEvent(payload));
+
+				async function recordEvent(payload) {
```

Comment:
This function is not really asynchronous and doesn't solve the problem with memory exhaustion. The CLI should run validate and/or write to console when individual records get ready.